### PR TITLE
Adding APR to boost import modal

### PR DESCRIPTION
**Pull request by IanWoodard**
image

### CODE REVIEW

File: earn/src/pages/BoostPage.tsx [hunk 1]
```diff
@@ -227,14 +254,16 @@ export default function BoostPage() {
...
       </div>
-      <ImportModal
-        isOpen={selectedPosition !== null && selectedCardInfo?.cardType === BoostCardType.UNISWAP_NFT}
-        uniqueId={selectedCardInfo ? getUniqueId(selectedCardInfo) : ''}
-        setIsOpen={() => {
-          setSelectedPosition(null);
-        }}
-        cardInfo={selectedCardInfo}
-      />
+      {selectedCardInfo !== undefined && (
```
Comment (haydenshively) [inline earn/src/pages/BoostPage.tsx:257]: Why is this necessary for the ImportModal but not the ManageBoostModal?

File: earn/src/data/Uniboost.ts [hunk 2]
```diff
@@ -36,6 +36,8 @@ export class BoostCardInfo {
     public readonly currentTick: number,
     public readonly token0: Token,
     public readonly token1: Token,
+    public readonly lender0: Address,
+    public readonly lender1: Address,
```
Comment (haydenshively) [inline earn/src/data/Uniboost.ts:40]: By adding these to the class, are you generating extra re-renders? i.e., all graphs are rendering as before and liquidation thresholds are calculated as before... but then they're rendered/calculated _again_ once `lender0` and `lender1` come in?

Comment (haydenshively) [reply earn/src/data/Uniboost.ts:40]: If this is the case, we should track lender0 and lender1 in some other data type.

File: earn/src/components/boost/ImportModal.tsx [hunk 3]
```diff
@@ -138,15 +150,43 @@ export default function ImportModal(props: ImportModalProps) {
...
+      );
+      if (isMounted) {
+        setMarketInfo(marketInfo);
+      }
+    }
+    fetchMarketInfo();
+    return () => {
+      isMounted = false;
+    };
+  }, [activeChain.id, cardInfo, provider, setMarketInfo]);
```
Comment (haydenshively) [inline earn/src/components/boost/ImportModal.tsx:188]: This could probably be done with wagmi hooks, but not a big deal either way.

**Review by haydenshively — APPROVED** [review]

File: earn/src/pages/BoostPage.tsx (hunk 1 above)
Comment (IanWoodard) [reply earn/src/pages/BoostPage.tsx:257]: I haven't updated the `ManageBoostModal` yet. It just allows us to not have undefined `cardInfo` in the modal.

File: earn/src/components/boost/ImportModal.tsx (hunk 3 above)
Comment (IanWoodard) [reply earn/src/components/boost/ImportModal.tsx:188]: I am just re-using old code here.

File: earn/src/data/Uniboost.ts (hunk 2 above)
Comment (IanWoodard) [reply earn/src/data/Uniboost.ts:40]: Good catch

Comment (IanWoodard) [reply earn/src/data/Uniboost.ts:40]: Actually, the way we currently do things, there should be no additional chart renders as we have a guard clause that ensures marketData is fetched before constructing the BoostCardInfos.
